src/ethereum.mapping.ts

```diff
@@ -91,103 +90,106 @@ export function handleCreateSplitCall(call: CreateSplitCall): void {
+  // TODO: explore cleaning this up w union type
+  let timestamp = call.block.timestamp;
+  let txHash = call.transaction.hash.toHexString();
   let splitId = call.inputs.split.toHexString();
   let tokenId = call.inputs.token.toHexString();
   let distributorAddress =
     call.inputs.distributorAddress != Address.zero()
       ? call.inputs.distributorAddress
       : call.from;
-  let txHash = call.transaction.hash.toHexString();
 
-  let amount = _getDistributionAmount(splitId, tokenId, txHash);
-  distributeSplit(splitId, tokenId, amount, distributorAddress);
+  let distributionEvent = _getDistributionEvent(
+    txHash,
+    splitId,
+    tokenId
+  ) as DistributionEvent;
+  let amount = distributionEvent.amount;
+  let logIdx = distributionEvent.logIndex;
+  distributeSplit(
+    timestamp,
+    txHash,
+    logIdx,
+    splitId,
+    tokenId,
+    amount,
+    distributorAddress
+  );
 }
 
 export function handleDistributeETHCall(call: DistributeETHCall): void {
+  // TODO: explore cleaning this up w union type
+  let timestamp = call.block.timestamp;
+  let txHash = call.transaction.hash.toHexString();
   let splitId = call.inputs.split.toHexString();
   let tokenId = Address.zero().toHexString();
   let distributorAddress =
     call.inputs.distributorAddress != Address.zero()
       ? call.inputs.distributorAddress
       : call.from;
-  let txHash = call.transaction.hash.toHexString();
 
-  let amount = _getDistributionAmount(splitId, tokenId, txHash);
-  distributeSplit(splitId, tokenId, amount, distributorAddress);
+  let distributionEvent = _getDistributionEvent(
+    txHash,
+    splitId,
+    tokenId
+  ) as DistributionEvent;
+  let amount = distributionEvent.amount;
+  let logIdx = distributionEvent.logIndex;
+  distributeSplit(
+    timestamp,
+    txHash,
+    logIdx,
+    splitId,
+    tokenId,
+    amount,
+    distributorAddress
+  );
 }
 
-function _getDistributionAmount(
+function _getDistributionEvent(
+  txHash: string,
   splitId: string,
-  tokenId: string,
-  txHash: string
-): BigInt {
+  tokenId: string
+): DistributionEvent | null {
   // must exist (event handlers fire before call handlers)
   let tx = Transaction.load(txHash) as Transaction;
   // must exist (event handlers fire before call handlers)
   let distEvents = tx.distributionEvents as Array<string>;
-  let amount = ZERO;
-  let logIdx = MAX_BIGINT;
+  let distEvent: DistributionEvent;
   for (let i = 0; i < distEvents.length; i++) {
     let distEvent = DistributionEvent.load(distEvents[i]) as DistributionEvent;
     // take the earliest event that exists matching the split & token
-    if (
-      distEvent.split == splitId &&
-      distEvent.token == tokenId &&
-      distEvent.logIndex < logIdx
-    ) {
-      amount = distEvent.amount;
-      logIdx = distEvent.logIndex;
+    // note: if we want to support txns that distribute the same token for the
+    // same split twice, will need to add some kind of 'processed' boolean to
+    // event
```

Comment:
could that even happen within the same txn?

Reply:
i mean i think it **could**.. i'm not sure it would make any sense tho (maybe someone wants to like distribute and then updateAndDistribute back to back??). it seems unlikely enough that i didn't build around it, but i did want to note it in case it caused problems in the future

Reply:
got it, ok cool